tests/shared/s3-packages.spec.js

```diff
@@ -0,0 +1,1012 @@
+          m.chai.expect(latestVersion).to.equal('1.0.0');
+          done();
+        }).catch(done);
+      });
+
+    });
+
+    describe('given production v1, v2, and v3 remote versions', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '3.0.1',
+          '3.0.0',
+          '2.1.1',
+          '2.1.0',
+          '2.0.0',
+          '1.2.0',
+          '1.1.0',
+          '1.0.2',
+          '1.0.1',
+          '1.0.0'
+        ]));
+      });
+
+      afterEach(function() {
+        this.getRemoteVersionsStub.restore();
+      });
+
+      it('should be able to resolve the latest v1 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '<2.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('1.2.0');
+          done();
+        }).catch(done);
+      });
+
+      it('should be able to resolve the latest v2 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '>=2.0.0 <3.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('2.1.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should be able to resolve the latest v3 version with a semver range', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          range: '>=3.0.0'
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('3.0.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should resolve the latest version if useUnstableChannel is true', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          useUnstableChannel: true
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('3.0.1');
+          done();
+        }).catch(done);
+      });
+
+      it('should resolve the latest version if useUnstableChannel is false', function(done) {
+        s3Packages.getLatestVersion(release.RELEASE_TYPE.PRODUCTION, {
+          considerUnstableReleases: false
+        }).then((latestVersion) => {
+          m.chai.expect(latestVersion).to.equal('3.0.1');
+          done();
+        }).catch(done);
+      });
+    });
+
+    describe('given pre-release production remote versions', function() {
+
+      beforeEach(function() {
+        this.getRemoteVersionsStub = m.sinon.stub(s3Packages, 'getRemoteVersions');
+        this.getRemoteVersionsStub.returns(Bluebird.resolve([
+          '2.0.0-beta.3',
+          '2.0.0-beta.2',
+          '2.0.0-beta.1',
+          '2.0.0-beta.0',
+          '1.0.0-beta.19',
+          '1.0.0-beta.18',
+          '1.0.0-beta.17',
+          '1.0.0-beta.16'
```

Comment:
I guess we should also have a test with a bucket containing both stable and unstable production versions, and checking that `useUnstableChannel` works as expected. In fact, there should probably be an extra two buckets to test - one where the latest version is a stable production release, and one where the latest version is an unstable production release. (so an extra 4 tests in total)

(You _could_ also do the same for snapshot releases, but I think that should be unnecessary, given the number of other tests we already have?)